bokeh/core/properties.py

```diff
@@ -1428,14 +1456,31 @@ def validate(self, value):
     def __str__(self):
         return "%s(%s)" % (self.__class__.__name__, ", ".join(map(repr, self.allowed_values)))
 
+    def _sphinx_type(self):
+        # try to return a link to a proper enum in bokeh.core.enums if possible
+        if self._enum in enums.__dict__.values():
+            for name, obj in enums.__dict__.items():
+                if self._enum is obj:
+                    val = _MODEL_LINK % "%s.%s" % (self._enum.__module__, name)
+        else:
+            val = str(self._enum)
+        return _PROP_LINK % self.__class__.__name__ + "( %s )" % val
+
 class Auto(Enum):
+    """ Accepts the string "auto".
 
+    Useful for properties that can be configured to behave "automatically".
+
+    """
     def __init__(self):
         super(Auto, self).__init__("auto")
 
     def __str__(self):
         return self.__class__.__name__
 
+    def _sphinx_type(self):
+        return _PROP_LINK % self.__class__.__name__
```

Comment:
@bryevdv, I see this duplicated a lot. Can't this be moved to appropriate super-class and overriden only when necessary?